### PR TITLE
NotificationDetails + Swift Mk1: Interface Initialization

### PR DESCRIPTION
**Pull request by jleandroperez**
#### Details:

Migrating **NotificationDetailsViewController**, step by step!.
Reference #5613

Needs Review: @kwonye 
Sir, you game for a (hopefully fun) review?

**Thanks a lot!!!**

--

#### Scenario A: TableView Cells
1. Log into a dotcom account
2. Open the Notifications Tab
3. Open a `Likes` Notification
3.1. Verify that the Header / User Cells are properly displayed
4. Open a `Badge` Notification
4.1. Verify that the Image / Text shows up correctly
4.2. Verify that it shows up vertically centered
5. Open a `Comment` Notification
5.1. Verify that the Header / Comment / Actions show up properly

Thanks!

#### Scenario B: Mark as Read
1. Log into a dotcom account
2. Open the Notifications Tab
3. Open any unread notification
4. Hit back, and please, verify that it remains marked as read

(Again, thanks!!)

#### Scenario C: Suggestions
1. Log into a dotcom account
2. Open the Notifications Tab
3. Open a Comment-Y Notification (nope, a Like won't do it)
4. Press over the ReplyTextView
5. Enter the `@` keyword + any character!

Please, verify that the Suggestions tableView shows up onscreen, with the possible handles.

#### Scenario D: Comment(s) Keyboard Dismissal
1. Log into a dotcom account **on an iPhone Device**
2. Open the Notifications Tab
3. Open a Comment-Y Notification (nope, a Like won't do it)
4. Press over the ReplyTextView
5. Swipe down (like... iMessage's App?), please, attempt an `Interactive dismiss`, and verify that everything works as expected
6. Once the keyboard is down, please, press over the ReplyTextView again
7. Please, press over the Comment area itself. Verify that the keyboard gets dismissed (we're testing the `Gestures Recognizer`!)

Please, note that the `ReplyTextView` component should not be visible on iPad Devices.

#### Scenario E: Notification Update
1. Receive a Comment Notification, for a testing blog
2. Open the Notifications tab (iOS)
3. Open the Notification's details
4. On the web, trash the comment

Please, verify that iOS immediately receives the deletion, and pop's the Details view.

### CODE REVIEW

File: WordPress/Classes/Models/Notifications/Notification.h [hunk 1]
```diff
@@ -64,47 +64,47 @@ typedef NS_ENUM(NSInteger, NoteBlockGroupType)
...
+@property (nonatomic, strong,  nonnull, readonly) NSDate                            *timestampAsDate;
+
+@property (nonatomic, assign, readonly) BOOL                                        isMatcher;
+@property (nonatomic, assign, readonly) BOOL                                        isComment;
+@property (nonatomic, assign, readonly) BOOL                                        isPost;
+@property (nonatomic, assign, readonly) BOOL                                        isFollow;
+@property (nonatomic, assign, readonly) BOOL                                        isLike;
+@property (nonatomic, assign, readonly) BOOL                                        isCommentLike;
+@property (nonatomic, assign, readonly) BOOL                                        isBadge;
+@property (nonatomic, assign, readonly) BOOL                                        hasReply;
```
Comment (kwonye) [inline WordPress/Classes/Models/Notifications/Notification.h]: I just died a little seeing this diff block #foreverunaligned

(kidding... kind of)

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+/// Setup Helpers
+///
+extension NotificationDetailsViewController
+{
+    func setupNavigationBar() {
+        // Don't show the notification title in the next-view's back button
+        navigationItem.backBarButtonItem = UIBarButtonItem(title: String(),
+                                                           style: .Plain,
+                                                           target: nil,
+                                                           action: nil)
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: I believe the style guide calls for aligning against the `:`

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+
+    func setupMainView() {
+        view.backgroundColor = WPStyleGuide.itsEverywhereGrey()
+    }
+
+    func setupTableView() {
+        tableView.separatorStyle            = .None
+        tableView.keyboardDismissMode       = .Interactive
+        tableView.backgroundColor           = WPStyleGuide.greyLighten30()
+        tableView.accessibilityIdentifier   = "Notification Details Table"
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: Please convert to NSLocalizedString

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+    }
+
+    func setupKeyboardManager() {
+        precondition(replyTextView != nil)
+        precondition(bottomLayoutConstraint != nil)
+
+        keyboardManager = KeyboardDismissHelper(parentView: view,
+                                                scrollView: tableView,
+                                                dismissableControl: replyTextView,
+                                                bottomLayoutConstraint: bottomLayoutConstraint)
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: Again, I believe we are supposed to align against the `:`

File: WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+    func setupMediaDownloader() {
+        // TODO: Nuke this method as soon as the Header is 100% Swift
+        mediaDownloader = NotificationMediaDownloader()
+    }
+
+    func setupReplyTextView() {
+        let replyTextView = ReplyTextView(width: view.frame.width)
+        replyTextView.placeholder = NSLocalizedString("Write a reply…", comment: "Placeholder text for inline compose view")
+        replyTextView.replyText = NSLocalizedString("Reply", comment: "").uppercaseString
+        replyTextView.accessibilityIdentifier = "Reply Text"
```
Comment (kwonye) [inline WordPress/Classes/ViewRelated/Notifications/Controllers/NotificationDetailsViewController+Extensions.swift]: `NSLocalizedString` please 😄

**Comment by kwonye** [comment]
I believe there is an issue with printing out an optional string with `Optional("<") You replied to this comment

simulator screen shot jul 27 2016 9 06 41 pm

**Comment by kwonye** [comment]
Besides the comments above, everything looks good. 

I trust you to address them as you see fit (if you disagree with the `:` alignment, etc, that's fine. Leave it as is). Once they're all addressed, :shipit:

File: WordPress/Classes/Models/Notifications/Notification.h (hunk 1 above)
Comment (jleandroperez) [inline WordPress/Classes/Models/Notifications/Notification.h]: Old habits die hard Will!

**Comment by jleandroperez** [comment]
Thanks *A LOT* for the review @kwonye (argh i almost poke, AGAIN, the wrong Will).

Re: Colons alignment, couldn't spot it in the Swift Guide!. Xcode suggests this alignment (CTRL + I over the code snippet), since we'd need to align space by space, let's shipit this way (i think that was the ObjC style, is that possible?)

Again, thanks!!!

**Pull request merged**